sunpy/net/dataretriever/sources/vsm.py

```diff
@@ -0,0 +1,163 @@
+    2015-01-03 00:00:00 2015-01-04 00:00:00  SOLIS        vsm
+    2015-01-04 00:00:00 2015-01-05 00:00:00  SOLIS        vsm
+    2015-01-05 00:00:00 2015-01-06 00:00:00  SOLIS        vsm
+    2015-01-06 00:00:00 2015-01-07 00:00:00  SOLIS        vsm
+    2015-01-07 00:00:00 2015-01-08 00:00:00  SOLIS        vsm
+    2015-01-08 00:00:00 2015-01-09 00:00:00  SOLIS        vsm]
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        table_physobs = ['EQUIVALENT_WIDTH', 'LOS_MAGNETIC_FIELD', 'VECTOR_MAGNETIC_FIELD']
+        table_wave = {6302:'v72', 8542:'v82', 10830:'v22'}
+
+        physobs_in = ('physobs' in kwargs.keys() and kwargs['physobs'] in table_physobs)
+##        url_pattern = 'http://gong2.nso.edu/pubkeep/{dtype}/%Y%m/k4{dtype}%y%m%d/k4{dtype}%y%m%dt%H%M%S{suffix}'
+
+        wave_float = kwargs['wavelength'].min.value
+
+        #The isclose function in numpy can only check two arrays if they
+        #are close to each other within some precision. Standalone values such
+        #as int, doubles etc can't be checked that way. In order to do that, we put the
+        #two indiviual values in two seperate arrays da and db and then apply
+        #isclose on both those arrays.
+        da = list()
+        da.append(wave_float)
+        for wave_nums in table_wave.keys():
+            db = list()
+            db.append(wave_nums)
+            if np.isclose(da, db, 1e-10, 1e-10):
+                wave = wave_nums
+                break
+
+        if wave is None:
+            raise ValueError("Enter correct wavelength values and units")
+
+        def download_from_nso(dtype, suffix, timerange):
+            tmp_pattern = 'k4{dtype}%y%m%dt%H%M%S{suffix}'
+            result = list()
+            base_url = 'http://gong2.nso.edu/pubkeep/{dtype}/{date:%Y%m}/k4{dtype}{date:%y%m%d}/'
+            total_days = (timerange.end - timerange.start).days + 1
+            all_dates = timerange.split(total_days)
+            for day in all_dates:
+                try:
+                    html = urllib2.urlopen(base_url.format(dtype=dtype, date=day.end))
+                    soup = BeautifulSoup(html)
+                    for link in soup.findAll("a"):
+                        crawler = Scraper(tmp_pattern, dtype=dtype, suffix=suffix)
+                        simple_pattern = '%y%m%d_%H%M%S'
+                        url = str(link.get('href'))
+                        if crawler._URL_followsPattern(url):
+                            tmp = re.sub('k4'+dtype, '', url)
+                            tmp = re.sub('[a-zA-Z]', '', tmp)
+                            crawler_ = Scraper(simple_pattern)
+                            if (timerange.start <= crawler_._extractDateURL(tmp) <= timerange.end):
+                                result.append(base_url.format(dtype=dtype, date=day.end)+url)
+                except:
```

Comment:
Do pass only the ones to do with no finding fits files, not everything else. If I break scraper in the future, I don't want this silently passing because the error is being `pass`